src/utils.ts

```diff
@@ -42,6 +43,12 @@ export function hasWarningDiagnostic(diagnostics: ISpectralDiagnostic[]): boolea
   return diagnostics.some(diagnostic => diagnostic.severity === DiagnosticSeverity.Warning);
 }
 
+export function setExtension(id: string, value: any, model: BaseModel): void {
+  id = id.startsWith('x-') ? id : `x-${id}`;
```

Comment:
We are adding a technical limitation here: extensions with the shape `x-x-whatever` are being transformed to `x-whatever`, i.e. `x-x-wing`. I know it might seem a remote edge case, but it is still valid. What if we just assume the user should provide the full extension name (including the `x-` prefix`) and we validate it is happening? Still it might happen the user sets `x-wing` expecting `x-x-wing` as extension, but at least there is no limitation.

Reply:
Yes 😄